api/src/Tests/Cases/RegistrationTest.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace App\Tests\TestCase\Controller;
+
+use App\Tests\Base\CiabTestCase;
+
+class RegistrationTest extends CiabTestCase
+{
+
+    private $ticket = null;
+
+
+    public static function setUpBeforeClass(): void
+    {
+        parent::setUpBeforeClass();
+        if (!array_key_exists('REMOTE_ADDR', $_SERVER)) {
+            $_SERVER['REMOTE_ADDR'] = '10.10.0.0';
+        }
+
+    }
+
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+    }
+
+
+    protected function tearDown(): void
+    {
+        if ($this->ticket) {
+            $this->runRequest('DELETE', '/registration/ticket/'.$this->ticket->id, null, null, 204);
+        }
+        parent::tearDown();
+
+    }
+
+
+    public function testOpen(): void
+    {
+        $this->runSuccessJsonRequest('GET', '/registration/open');
+        $this->runRequest('GET', '/registration/open/-1', null, null, 404);
+
+    }
+
+
+    public function testAdmin(): void
+    {
+        $this->runSuccessJsonRequest('GET', '/registration/admin');
+        $this->runSuccessJsonRequest('GET', '/registration/configuration');
+        $this->runRequest('GET', '/registration/configuration/nothing', null, null, 404);
+        $this->runRequest('PUT', '/registration/configuration', null, null, 400);
+        $this->runSuccessJsonRequest('PUT', '/registration/configuration', null, ['Field' => 'ForceOpen', 'Value' => '1']);
+
+    }
+
+
+    public function testTicket(): void
+    {
+        $this->runRequest('GET', '/registration/ticket/type/-1', null, null, 404);
+        $type = $this->runSuccessJsonRequest('GET', '/registration/ticket/type');
+        $data = $this->runSuccessJsonRequest('GET', '/registration/ticket/type/'.$type->id);
+        $this->assertEquals($type, $data);
+
+        $this->runRequest('POST', '/registration/ticket', null, null, 400);
```

Comment:
Could this block be done with a data provider?

Reply:
I tried but we used $type->id and the data provider it actually run before any of the class is initialized.  So we cannot hav any figured or dynamic values like the id from a type we query from the DB.  We could partly dry it up by maybe proving all BUT that value and having some flag that replaces a placeholder with the value, but that started to get convoluted and I was unsure if it was cleaner or not.

Reply:
OK, that's legit. I'll add contemplating that kind of DRYing up to the pile of background-thinking I'm doing over how to better dance with PHPUnit.